refactor(core-ui): Do Popover template with preact

klarstrup opened the pull request:
Preact is a rendering library with an API similar to and most of the basic features of React, despite being only slightly larger than Mustache. 

It will save us a lot of messing around with rendering Mustache templates into `innerHTML` and then trying to find the right elements to attach events to with `Gator.on` or whatever. 

It will also ease maintenance and codebase cohesion across Tjek by becoming more similar to Vizsla.


The first commit on this PR is an example of using it in a very local and contained way, to make an example that we can use this quite atomically.

Commit:
remove unused function

Commit:
Hammer prefixed use find

Commit:
Prefer common methods over loops

Commit:
hammer: expunge uniqueArray

Commit:
update deps

Commit:
mark @tjek/sdk sideEffects free

Commit:
avoid destructuring arrays

Commit:
avoid Object.prototype.hasOwnProperty.call

Commit:
Hammer: TouchInput fix getTouches

Commit:
tighten babel config

Commit:
update deps

Commit:
@tjek/sdk: dedrop sub-package nonsense

Commit:
update deps

jrlarano committed:
[Template] Incito viewer (#106)

* [Template] Initial commit for Incito viewer

* [Template] Add scroll to offer and section

* [Template] Add types to request and add preferred type

* [Template] Add Section list in incito

* Fix shopping list format for sharing

* [Incito template] Fix shopping list without price

* [Template Incito] Filter offers for incito

* [Tempate incito] Add loader and scroll progress level

* [template incito] Reuse shopping list from paged publication viewer

* [Template incito] Add progress percentage

* [Template Incito] Remove console logs

* [Template] Fix Shopping list print layout

* [Template][Refactor] Initial consolidation of PDF and Incito viewer

* [Template] Add optional attributes to pub list sample

* [Template] Use single container for PDF and Incito viewer

* [Template] Unify Header component

* [Template] Move header to common

* [Template] Optimize header for incito and paged

* [Template] Add tjek link to readme

* [Template] change header color based on brightness

* [Template] Consolidate menu-popup and offer list for incito and paged

* Fix lint error

* [Template Incito] Support custom template for section list

* [Template] Optimize header styling

* [Template] Fix header z-index

* [Template] Fix progress label

* [Template Incito] Fix webshop link on offer click

* [Template Incito] dispatch event on offer click

* [Template] Add offset to section scroll

* [Template Incito] Fetch offer on cell click

* [Template Incito] Add offer click tracking

* [Incito template] Move tracker to fetchOffer function

* [Template Incito] Refactor fetch offer param names

* Remove fetchOffer from incito Bootstrapper & don't make integrators make redundant fetchOffer call

* Fix template examples relative paths

* sort imports

Co-authored-by: klarstrup <jonas@klarstrup.dk>

jrlarano committed:
[Template] Fix Paged viewer overflow & print (#108)

* [Template] Fix Paged viewer overflow

* [Template] Add html class when publication viewer is open

* [Template] Rename class names

* [Template] Add option to disable global scroll

* [Template] Refactor Incito for global scrollbar

* [Template] Fix Print and remove sgn_publications max-width

Commit:
update deps

Commit:
Tracker: Avoid some polyfills

jrlarano committed:
Minor fixes (#109)

* Fix share btn showing after clicking from list

* [Template Incito] update fixed position, z-index, and bg color

* [Incito template] Refactor fixed header

* [Incito Tempalte] Add loader bg

* [Incito template] Fix open tab being blocked

* [Incito template] Fix click dispatch event

* [Incito template] Add open new tab fallback and minor adjustments

Commit:
update deps

Commit:
optimize incito-browser for ES5 transpilation

Commit:
incito-browser: guard Node use

Commit:
update deps

dependabot[bot] committed:
Bump async from 2.6.3 to 2.6.4 (#114)

Bumps [async](https://github.com/caolan/async) from 2.6.3 to 2.6.4.
- [Release notes](https://github.com/caolan/async/releases)
- [Changelog](https://github.com/caolan/async/blob/v2.6.4/CHANGELOG.md)
- [Commits](https://github.com/caolan/async/compare/v2.6.3...v2.6.4)

---
updated-dependencies:
- dependency-name: async
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump minimist from 1.2.5 to 1.2.6 (#110)

Bumps [minimist](https://github.com/substack/minimist) from 1.2.5 to 1.2.6.
- [Release notes](https://github.com/substack/minimist/releases)
- [Commits](https://github.com/substack/minimist/compare/1.2.5...1.2.6)

---
updated-dependencies:
- dependency-name: minimist
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump simple-get from 3.1.0 to 3.1.1 (#103)

Bumps [simple-get](https://github.com/feross/simple-get) from 3.1.0 to 3.1.1.
- [Release notes](https://github.com/feross/simple-get/releases)
- [Commits](https://github.com/feross/simple-get/compare/v3.1.0...v3.1.1)

---
updated-dependencies:
- dependency-name: simple-get
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

jrlarano committed:
Add option to display query params (#111)

* Add support to display query params on publication list

* [Template] Display page number on query params

* Add option to display pubId and pageId as url hash

* Fix lint errror

* Add optional chaining

* Refactor query param

* [Template] Update url based on progress label change event

* [Fix] Lint error

mortenbo committed:
Sanity check URLs and fix fonts. Not sure how it worked before

mortenbo committed:
Remove startsWith

mortenbo committed:
Use map

jrlarano committed:
Feat/offer info and minor enhancements (#113)

* Add support to display query params on publication list

* [Template] Display page number on query params

* Add option to display pubId and pageId as url hash

* Fix lint errror

* [Template] Add Labels below header icons

* Add optional chaining

* Refactor query param

* [Template] Add options to display header labels and offer overview modal

* [Template]Implement offer modal to paged viewer

* [Template]Improve offer overview modal

* [Template] Update url based on progress label change event

* [Fix] Lint error

* [Template] Add option to display controls going to first and last page

* [Template] Hide header control on mobile, update header label

* [Template] Allow only 1 line on pub list heading

* [Template] Inherit font-family

* [Template] Use offer date instead of pub date for modal heading

* [Template] Reposition and remove option to hide first/last icons

* [Template] Use mailto as Share list fallback

* [Template] Add fadeout animation in modal

* [Template] Add modal loader on paged offer overview

* [Template] Add modal loader on paged offer overview

* [Template] Use absolute path

* [Template] Minor code refactor

* [Template] Display Incito scroll bar and percent at the bottom if no header

mortenbo committed:
Dont allow same-origin sandbox

mortenbo committed:
Merge pull request #115 from shopgun/mr/sanity_check_urls

Sanity check URLs and fix fonts. Not sure how it worked before

jrlarano committed:
[Viewer] Minor styling fixes (#116)

* [Viewer] Minor styling fixes

* [Viewer] Revert bottom spacing

* [Viewer] Update download pdf

* [Viewer] Remove bottom progress if position is not fixed

Commit:
examples: use relative asset paths

jrlarano committed:
[Viewer]Remove manual hiding of nav icons (#117)

Commit:
update deps

Commit:
publish: add experimental release channel

jrlarano committed:
Feat/minor changes to viewer (#119)

* [Viewer] List publications in 2 cols

* [Viewer] bigger offer overview

* [Viewer] translate shopping list and remove [] on share option

* [Viewer] minor style changes

* [Viewer] Fix share bug on Safari

* [Viewer] offer list order by page

Commit:
update dependencies

Commit:
add CDN prefix to S3 upload script success output

jrlarano committed:
[Viewer] Fix progress label being cut (#120)

jrlarano committed:
[Incito] Reinstate allow-same-origin (#121)

Commit:
upload-s3 script: add cloudfront invalidation

Commit:
update deps

jrlarano committed:
[Viewer] Dispatch even on Incito load (#122)

* [Viewer] Dispatch even on Incito load

* Add plugin to support experimental syntax 'classPrivateProperties' and stop breaking the build

* Remove window event, add option for manual render with callback

* remove redundant

* [Viewer] Expose data via render promise

* [Viewer] Minor refactor

* [Viewer] dispatch element event on publication load

* [Viewer] Move events from windows to element

* [Viewer] Refactor event listeners

* [Viewer] Rename event name

* [Viewer] rename publication click event

Co-authored-by: t-lansing <tereza@lansing.dk>
Co-authored-by: Jevy Roque H. Laraño <210886.JLarano@gbs.local>

jrlarano committed:
[Viewer] Fix offer modal img width (#124)

Commit:
Migrate all source code to TypeSrcript without affecting output code (#118)

* ts

* ts

* update deps and revert module type

* add typescript lol

* CI: Drop node 12

* update deps

* migrate ui components to typescript

Commit:
Add stuff vizsla needs to @tjek/sdk

jrlarano committed:
Fix date and modal (#126)

* [Viewer] Add offer modal scroll

* [Viewer]Fix date format for Safari

* Remove console log

* [Viewer] Add date parser

jrlarano committed:
[Viewer] Fix css inset issue on safari (#127)

Commit:
update deps

jrlarano committed:
Use date range format on offer modal (#125)

* Use date range format on offer modal

* [Viewer] Add offer modal scroll

* [Viewer] Fix css inset issue on safari

* [Viewer] Add offer date range and offer status message

* Minor cleanup

* [Viewer] Update offer modal date range status

* [Viewer]Refactor offer date format

Commit:
hide misleading validity indicators

jrlarano committed:
Fix offer date color and status message (#128)

* Initial fix to date color and status message

* [Viewer]include timeZoneOffset in calculating offer dates

* Refactor to hex color

* [Viewer] Refactor date range and implement to all modals

* [Viewer] Cleanup

* Offer date message to compute using ceil instead of round

* [Viewer] Remove date message temporarily

* [Viewer] Refactor date value to always use Copenhagen timezone

* [Viewer] Remove fixed timezone

Commit:
readd missing hammerjs method

Commit:
update deps

jrlarano committed:
[Viewer]Dispatch event on offer modal rendered (#130)

jrlarano committed:
[Viewer] Support templates for date range (#129)

Commit:
update deps

Commit:
PagedPublicationKit: Add hotspotsPointerdown

Commit:
PagedPublicationKit: Add hotspotsPointerdown

Commit:
update contributors

mortenbo committed:
Make zoom and center customizable

mortenbo committed:
make radius customizable

jrlarano committed:
Disable pointer if offer is not clickable (#133)

jrlarano committed:
Refactor/offer cursor condition (#134)

* Disable pointer if offer is not clickable

* Improve offer clickable condition

Commit:
Add missing hammer Recognizer property

Commit:
Merge branch 'develop' of https://github.com/shopgun/shopgun-js-sdk into develop

Commit:
CI: Reduce build to the tested bundles

klarstrup committed:
Track paged publication open on open (#135)

Co-authored-by: klarstrup <io@klarstrup.dk>

Commit:
update deps

Commit:
update deps

dependabot[bot] committed:
Bump terser from 5.7.1 to 5.15.0 (#136)

Bumps [terser](https://github.com/terser/terser) from 5.7.1 to 5.15.0.
- [Release notes](https://github.com/terser/terser/releases)
- [Changelog](https://github.com/terser/terser/blob/master/CHANGELOG.md)
- [Commits](https://github.com/terser/terser/compare/v5.7.1...v5.15.0)

---
updated-dependencies:
- dependency-name: terser
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump shell-quote from 1.7.2 to 1.7.3 (#123)

Bumps [shell-quote](https://github.com/substack/node-shell-quote) from 1.7.2 to 1.7.3.
- [Release notes](https://github.com/substack/node-shell-quote/releases)
- [Changelog](https://github.com/substack/node-shell-quote/blob/master/CHANGELOG.md)
- [Commits](https://github.com/substack/node-shell-quote/compare/v1.7.2...1.7.3)

---
updated-dependencies:
- dependency-name: shell-quote
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

Commit:
update deps

Commit:
update readme

klarstrup committed:
Tjek where it makes sense

jrlarano committed:
Update README.md

dependabot[bot] committed:
Bump file-type from 16.5.3 to 16.5.4 (#132)

Bumps [file-type](https://github.com/sindresorhus/file-type) from 16.5.3 to 16.5.4.
- [Release notes](https://github.com/sindresorhus/file-type/releases)
- [Commits](https://github.com/sindresorhus/file-type/compare/v16.5.3...v16.5.4)

---
updated-dependencies:
- dependency-name: file-type
  dependency-type: direct:development
...

Signed-off-by: dependabot[bot] <support@github.com>

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump yargs-parser from 7.0.0 to 21.0.1 (#138)

Bumps [yargs-parser](https://github.com/yargs/yargs-parser) from 7.0.0 to 21.0.1.
- [Release notes](https://github.com/yargs/yargs-parser/releases)
- [Changelog](https://github.com/yargs/yargs-parser/blob/main/CHANGELOG.md)
- [Commits](https://github.com/yargs/yargs-parser/compare/v7.0.0...yargs-parser-v21.0.1)

---
updated-dependencies:
- dependency-name: yargs-parser
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

klarstrup committed:
update deps

klarstrup committed:
update deps

klarstrup committed:
update deps overrides

klarstrup committed:
downgrade jest

pending replacement of jest-playwright-preset with @playwright/test

klarstrup committed:
EventTracker: add eventsTrackHeaders option (#139)

* EventTracker: add eventsTrackHeaders option

* Tracker: Use plain object for eventsTrackHeaders

Seems crossfetch isn't playing nice with Headers objects

klarstrup committed:
PagedPublicationPageSpread: Add loadImage error text

jrlarano committed:
Page Decoration (#141)

* [PageDecoration] Initial commit

* revert viewer

* Add page decorations to core beforenav

* Add pagedecoration to pagespread

* Rename page decoration el

* Refactor implementation

* Fix stacked up decoration

* position on top, hide on zoomedin

* minor refactor

* Update readme

* Refactor implementation

* Reposition decoration and add simple animation

* cleanup

* Page decor update style

* [Page decors] Improve shadow

* Update page link positions

* Move Page links to bottom

jrlarano committed:
Fix the use pageSpreadEls width (#143)

klarstrup committed:
update deps

klarstrup committed:
update deps(typescript)

klarstrup committed:
update deps(jest)

klarstrup committed:
update deps(rollup plugins)

klarstrup committed:
update deps(rollup)

jrlarano committed:
Fix/page link position (#145)

* Fix page link position getting centered on double page

* [PageLink] Display subdomain if it's not www

jrlarano committed:
Add filter on viewer level (#144)

klarstrup committed:
convert main module and scripts to ESM

klarstrup committed:
fix upload script

klarstrup committed:
Verso: Update input class when pointer type changes

Fixes #148

klarstrup committed:
Verso: Update input class when pointer type changes

Fixes #148

klarstrup committed:
Support older MediaQueryList implementation

mortenbo committed:
Leverage browser lazy loading when possible for massive performance and UX improvements

mortenbo committed:
Merge pull request #150 from tjek/mr/lazy_loading

Leverage browser lazy loading when possible for massive performance and UX improvements

klarstrup committed:
update deps

mortenbo committed:
bugfix: Incito video loading

mortenbo committed:
Better lazy loading

klarstrup committed:
update deps

dependabot[bot] committed:
Bump mem and lock-verify (#147)

Removes [mem](https://github.com/sindresorhus/mem). It's no longer used after updating ancestor dependency [lock-verify](https://github.com/npm/lock-verify). These dependencies need to be updated together.


Removes `mem`

Updates `lock-verify` from 2.2.1 to 2.2.2
- [Release notes](https://github.com/npm/lock-verify/releases)
- [Commits](https://github.com/npm/lock-verify/commits)

---
updated-dependencies:
- dependency-name: mem
  dependency-type: indirect
- dependency-name: lock-verify
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump json5 from 2.2.1 to 2.2.3 (#152)

Bumps [json5](https://github.com/json5/json5) from 2.2.1 to 2.2.3.
- [Release notes](https://github.com/json5/json5/releases)
- [Changelog](https://github.com/json5/json5/blob/main/CHANGELOG.md)
- [Commits](https://github.com/json5/json5/compare/v2.2.1...v2.2.3)

---
updated-dependencies:
- dependency-name: json5
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump qs from 6.5.2 to 6.5.3 (#149)

Bumps [qs](https://github.com/ljharb/qs) from 6.5.2 to 6.5.3.
- [Release notes](https://github.com/ljharb/qs/releases)
- [Changelog](https://github.com/ljharb/qs/blob/main/CHANGELOG.md)
- [Commits](https://github.com/ljharb/qs/compare/v6.5.2...v6.5.3)

---
updated-dependencies:
- dependency-name: qs
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump got and lock-verify (#146)

Removes [got](https://github.com/sindresorhus/got). It's no longer used after updating ancestor dependency [lock-verify](https://github.com/npm/lock-verify). These dependencies need to be updated together.


Removes `got`

Updates `lock-verify` from 2.2.1 to 2.2.2
- [Release notes](https://github.com/npm/lock-verify/releases)
- [Commits](https://github.com/npm/lock-verify/commits)

---
updated-dependencies:
- dependency-name: got
  dependency-type: indirect
- dependency-name: lock-verify
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

klarstrup committed:
Merge branch 'develop' into mr/fix_video_issues

mortenbo committed:
Remove use of CloudFront so we're cloud agnostic

mortenbo committed:
Merge pull request #151 from tjek/mr/fix_video_issues

bugfix: Incito video loading

mortenbo committed:
Merge pull request #153 from tjek/mr/new_domain

Remove use of CloudFront so we're cloud agnostic

mortenbo committed:
Reduce delay to 0 if reduce motion is preferred

mortenbo committed:
Listen to events

mortenbo committed:
Boolean cast

mortenbo committed:
Merge pull request #154 from tjek/mr/reduce_motion

Reduce delay to 0 if reduce motion is preferred

klarstrup committed:
Verso: Improve portability of prefers-reduced-motion detection

klarstrup committed:
update deps

klarstrup committed:
update deps

klarstrup committed:
Verso: Simplify reduced motion compliance
Instead of listening and tracking when the reduced motion query changes, check it directly when we need to know. This only happens once per navigateTo call

mortenbo committed:
Simplify video autoload for Incito (#155)

* Simplify video autoload for Incito

* Remove default value

* Guard when incito video is loading

* better to check instanceof than compare tagNames

Co-authored-by: klarstrup <io@klarstrup.dk>

klarstrup committed:
update deps

jrlarano committed:
[Template]Fix hidden body on print media (#156)

* [Template]Fix hidden body on print media

* [Template]Remove afterprint event upon closing modal

jrlarano committed:
Refactor/shopping list print (#157)

* [Template]Support shopping list print via keyboard shortcut

* [Template] Refactor shopping list print

jrlarano committed:
Add header icon style to avoid being overruled

mortenbo committed:
Merge pull request #158 from tjek/refactor-header-icon-style

Add header icon style to avoid being overruled

klarstrup committed:
Include page decorations in basic paged-publication example

klarstrup committed:
Include initial page query param in basic paged-publication example

klarstrup committed:
Paged Publication Viewer: Render decorations immediately when applied (#160)

fixes #159

klarstrup committed:
improve PagedPublicationKit types

klarstrup committed:
Merge branch 'develop' of https://github.com/tjek/shopgun-js-sdk into develop

klarstrup committed:
update deps

mortenbo committed:
Improve Incito performance with async decoding

mortenbo committed:
Merge pull request #162 from tjek/mr/incito_image_loading

Improve Incito performance with async decoding

mortenbo committed:
Remove Pulse

klarstrup committed:
Incito Publication Kit Bootstrapper: Bind correct Incito `destroyed` event

klarstrup committed:
EventsKit: type Tracker.createViewToken

klarstrup committed:
EventsKit: type Tracker counstructor

klarstrup committed:
EventsKit: type trackEvent et. al.

klarstrup committed:
EventsKit: Avoid type assertion and improve inference

klarstrup committed:
ci(rollup): Avoid building bundles irrelevant to testing in CI

klarstrup committed:
Vendor and port MicroEvent to typescript

And give it generics powers to declare triggerable/bindable events

klarstrup committed:
MicroEvent: Add utility for getting event arguments

klarstrup committed:
update deps

klarstrup committed:
tests(webpack): Fix webpack test

klarstrup committed:
[Incito Reader, Event Tracking] Implement "Incito Publication Section Opened" event (#165)

* feat(incito-browser): Add sectionVisible and sectionHidden events

Also correct usage for the other two events and type all of the Incito MicroEvent events

* docs(incito-browser): Add sectionVisible and sectionHidden events debugging example

* docs(incito-browser): Clean up sectionVisible and sectionHidden events implementation

* EventsKit Tracker: Add trackIncitoPublicationSectionOpened

* Incito Publication Kit: Use trackIncitoPublicationSectionOpened

* EventsKit: fix trackIncitoPublicationOpened signature

* Incito Publication Kit/incito-browser: Clean types

klarstrup committed:
update bundlesize limits

klarstrup committed:
update deps

klarstrup committed:
fix(CoreUI transformIncitoOffer): Handle V4 validity range (#168)

* fix(CoreUI transformIncitoOffer): Handle V4 validity range

@jrlarano I suggest moving forward we try to parse dates as early as possible(i.e. immediately after fetching) instead of passing around string dates that are tedious to modify for purpose(as seen here).

* remember that import

klarstrup committed:
update bundlesize limits

klarstrup committed:
feat(@tjek/sdk): Export IncitoPublicationBootstrapper

klarstrup committed:
update deps

klarstrup committed:
docs(examples): IE11 friendly syntax

klarstrup committed:
fix(Hammer): Fix safari 10 crash

klarstrup committed:
refactor(PagedPublicationPageSpread): Drop cheeky Object.assign

klarstrup committed:
refactor(Config): Drop private field usage

This requires a WeakMap polyfill which isn't worth it

dependabot[bot] committed:
Bump @sideway/formula from 3.0.0 to 3.0.1 (#166)

Bumps [@sideway/formula](https://github.com/sideway/formula) from 3.0.0 to 3.0.1.
- [Release notes](https://github.com/sideway/formula/releases)
- [Commits](https://github.com/sideway/formula/compare/v3.0.0...v3.0.1)

---
updated-dependencies:
- dependency-name: "@sideway/formula"
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

klarstrup committed:
refactor(Verso): Use vendored MicroEvent

klarstrup committed:
feat(publish-npm): add DRY_RUN env var mode

klarstrup committed:
feat(publish-npm): add gh release tagging

klarstrup committed:
refactor(core-ui): Do Popover template with preact

klarstrup committed:
refactor(PagedPublicationKit): Omit Mustache.render

klarstrup committed:
chore(rollup): add missing preact external

klarstrup commented:
Next I would hope to test this out on one of the core-ui template components. 

Though this might be a non-starter if we have already committed to supporting custom Mustache templates for all those components, as is supported in the code. Do you know if anyone actually uses that @jrlarano? Who requested that feature?

klarstrup committed:
refactor(core-ui): Do Popover template with preact component

jrlarano commented:
@klarstrup this is really cool and clean 💪 If I remember correctly, it was @mortenbo's idea to use templating in every component so that the clients can override our default template and use their custom template.
eg:
https://github.com/tjek/tjek-js-sdk/blob/93aac1301c2f38c3f38423d9e5794ca43773610a/examples/list-publications-template.html#L26

I don't think there's any client that's using it atm but I'm not certain about that.

jrlarano committed:
Component style adjustments (#169)

* Change price color to black

* Contain offer image

klarstrup committed:
perf(browserslist): Drop IE11 support (#170)

Remove aggressive automatic polyfill adding and increase browserslist support level to supports css-grid, giving us this base level of browser support: https://caniuse.com/css-grid

klarstrup committed:
Merge branch 'develop' into io/preact

klarstrup committed:
chore(types): type hammer prefixed

klarstrup committed:
fix(publish-npm): Fix changed display

klarstrup committed:
chore(upload-s3): Handle multi-digit version numbers

klarstrup committed:
chore(publish-npm): Improve publish log

klarstrup committed:
chore(types): type hammer TouchAction

klarstrup committed:
Merge branch 'develop' into io/preact

klarstrup committed:
fix(PagedPublicationKit): Fix spread page decorations (#173)

klarstrup committed:
Merge branch 'develop' into io/preact

klarstrup committed:
types(V2PageDecoration): title and website_link are nullable

klarstrup committed:
Merge branch 'develop' into io/preact

klarstrup committed:
refactor(core-ui): Do PageDecorations template with preact

klarstrup commented:
Alright, I would like to get rid of them until we have a concrete need - then we can invent something then.

klarstrup committed:
tests(EventTracker): Make sure Wolf gets POSTed to

klarstrup committed:
update deps

klarstrup committed:
fix(sgn-sdk.js): Autoconfigure from current script tag only

Fixes #175

klarstrup committed:
chore(deps): Use new modular AWS SDK

jrlarano committed:
Add fontfamily and color to spans (#174)

* Add fontfamily and color to spans

* Refactor spans font to inherit

* Spans to inherit color

mortenbo committed:
Rejigger IncitoEmbedView to actually POST JSON

mortenbo committed:
Add incitoEmbedLoaded event

mortenbo committed:
fetch with timeout

mortenbo committed:
Enable enable lazy load param

klarstrup committed:
chore(deps): update deps

jrlarano committed:
Refactor: Incito and List containers to use preact (#178)

jrlarano committed:
Feat/sidebar menu (#142)

* Initial commit for sidebar

* Add sidebar component

* Update sidebar layout

* Add an option for right sidebar position

* sidebar default position: right

* Refactor sidebar style

* [Sidebar] Hide on mobile

* [Sidebar]Hide/show control improvement

* center modal on enabled sidebar

* Add fixed width on sidebar

* [Sidebar] Change control icon and animation

* [Sidebar] Implement to paged

* [Sidebar] update modal layout on mobile

* [Sidebar] Adjust control size

* [Sidebar] Hide pages modal

* [Sidebar Refactor] Move component inside header

* [Sidebar]Reduce width

* [Sidebar] Display page decorations if available

* [Sidebar]Add page decorations component

* [Sidebar] Minor style

* [Sidebar]Fix disable header

* [Sidebar] use matchMedia for screen resize

* minor function rename

* [Sidebar] Fix fixed position with overflow issue in Safari

mortenbo committed:
Merge pull request #176 from tjek/mr/embed_incito

Rejigger IncitoEmbedView to actually POST JSON

klarstrup committed:
fix(page-decorations): check website_link directly

klarstrup committed:
refactor(sgn-sdk): use named exports

klarstrup committed:
chore(deps): Declare direct glob dependency

klarstrup committed:
chore(scripts): npm run test in parallel

klarstrup committed:
chore(scripts): add test:types npm script

klarstrup committed:
ci(test): run test:types in CI

klarstrup committed:
chore(bundlesize): increase sgn-sdk.js size limit

klarstrup committed:
chore(ci): use node 18

klarstrup committed:
chore(rollup): don't skip bundles in CI

klarstrup committed:
chore(lint): allow importAssertions

klarstrup committed:
chore(rollup): don't skip bundles in CI

klarstrup committed:
chore(build): allow rollup more heap

klarstrup committed:
chore(build): allow rollup more heap

klarstrup committed:
chore(deps): update deps

klarstrup committed:
chore(CI): Simplify github job

klarstrup committed:
chore(ci): split up checks

klarstrup committed:
chore(ci): fix split up checks

klarstrup committed:
chore(ci): enable actions/setup-node caching

klarstrup committed:
style(whitespace): add a linebreak

klarstrup committed:
update deps

klarstrup committed:
Merge branch 'develop' into io/preact

klarstrup committed:
chore(jest): use jest-esbuild over ts-jest

klarstrup committed:
Merge branch 'develop' into io/preact

klarstrup committed:
tests(EventsKits): Be more patient for dispatch

klarstrup committed:
Merge branch 'develop' into io/preact

klarstrup committed:
fix(core-ui/incito-publication): fix prop name

klarstrup committed:
fix(core-ui/incito-publication): fix prop type

klarstrup committed:
chore(bundlesize): increase limit

klarstrup committed:
chore(bundlesize): increase limit

klarstrup committed:
chore(deps): update deps

klarstrup committed:
chore(rollup): add preact/hooks external

klarstrup committed:
chore(publish-npm): updated glob API

klarstrup committed:
chore(upload-s3): set AWS region

klarstrup committed:
chore(deps): update deps

klarstrup committed:
refactor(Incito Kit Viewer): remove unused id option

klarstrup committed:
refactor(tests): Typescripty tests

klarstrup committed:
tests(incito-browser): ignore type mismatch with incito fixture

klarstrup committed:
chore(deps): update deps

klarstrup committed:
chore(deps): update deps(typescript)

klarstrup committed:
Merge branch 'develop' into io/preact

klarstrup committed:
chore(deps): remove jest-test-matrix

klarstrup committed:
tests(refactor): use wildcard imports for SGN tests

klarstrup committed:
chore(lint): Use @typescript-eslint for real good type checking

klarstrup committed:
chore(deps): Declare node-gyp version

Commit:
Feat/set lazyload in constructor (#182)

klarstrup committed:
Merge branch 'develop' of https://github.com/tjek/tjek-js-sdk into develop

klarstrup committed:
chore(CI): Require biuld for test:types

klarstrup committed:
chore(CI): Require biuld for test:types

klarstrup committed:
chore(deps): update deps

klarstrup committed:
chore(mjs): Drop type module and use mjs for scripts

klarstrup committed:
chore(deps): Obviate jest-each

klarstrup committed:
chore(scripts): include non-minified assets on S3

klarstrup committed:
feat(Tracker): Add location source setting (#185)

fixes #184

klarstrup committed:
chore(GitHub CI): Cache playwright binaries

Also only install chromium

klarstrup committed:
chore(GitHub CI): Really only install chromium

klarstrup committed:
chore(GitHub CI): remove unnecessary playwright step

klarstrup committed:
Merge branch 'develop' into io/preact
